### PR TITLE
[Refactor] Provide a common module for sharing between chat / calling

### PR DESCRIPTION
**Pull request by emlynmac**
## Purpose
This PR shows how to use SPM internally to move common functionality into a package, with an SPI include, to enable code sharing between packages.

We can split out the common areas between chat and calling as we find them and place them into this package.

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Pull and build; run the demo app

## What to Check
No changes; no breaks.

## OTHER NOTES
Make sure you have your ruby gem `xcodeproj` updated to latest (1.22 currently)

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Manager/AudioSessionManager.swift [hunk 1]
```diff
@@ -4,6 +4,7 @@
 //
 
 import AVFoundation
+@_spi(common) import AzureUICommunicationCommon
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Manager/AudioSessionManager.swift:7]: hey do we need to define this keyword "common" somewhere so swift would know which package we are importing from? For chat, would this be simply @ _spi(chat)?

Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Manager/AudioSessionManager.swift:7]: The identifier is whatever we choose it to be. We can have different `@_spi` exported and imported; I just used common as it is common code.

File: AzureCommunicationUI/sdk/AzureCommunicationUICommon/.swiftpm/xcode/xcshareddata/xcschemes/AzureUICommunicationCommon.xcscheme [hunk 2]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICommon/.swiftpm/xcode/xcshareddata/xcschemes/AzureUICommunicationCommon.xcscheme:1]: It looks like the file name should be AzureCommunicationUICommon?

File: AzureCommunicationUI/sdk/AzureCommunicationUICommon/AzureCommunicationUICommon.podspec [hunk 3]
```diff
@@ -0,0 +1,19 @@
+Pod::Spec.new do |s|
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICommon/AzureCommunicationUICommon.podspec:1]: I'm wondering if we still need this file as we won't be used any private pods

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/DI/DependencyContainer.swift [hunk 4]
```diff
@@ -27,7 +27,7 @@ final class DependencyContainer {
     }
 
     private func registerDefaultDependencies() {
-        register(DefaultLogger() as Logger)
+        register(DefaultLogger(category: "Calling") as Logger)
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/DI/DependencyContainer.swift:30]: does it make sense to have a enum for category? Just feel like this is bit of error prone due to typos and such

File: AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/CancelBag.swift [hunk 5]
```diff
@@ -6,6 +6,9 @@
 import Combine
 
 final class CancelBag {
+
+    init() { }
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/CancelBag.swift:10]: is this necessary?

File: AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/StringConstants.swift [hunk 6]
```diff
@@ -5,7 +5,7 @@
 
 import Foundation
 
-struct StringConstants {
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/StringConstants.swift:8]: good catch here!

**Review by jpeng-ms — COMMENTED** [review]
it seems we left out some providers in common. Is this intentional?

File: AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/CancelBag.swift (hunk 5 above)
Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/CancelBag.swift:10]: It was originally as this was at one point a @_spi public interface, so the initializer needed to be there too.
I can remove it for now.

File: AzureCommunicationUI/sdk/AzureCommunicationUICommon/AzureCommunicationUICommon.podspec (hunk 3 above)
Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICommon/AzureCommunicationUICommon.podspec:1]: I can remove it for now

**Comment by emlynmac** [comment]
> it seems we left out some providers in common. Is this intentional?

What are you referring to here?

**Comment by jpeng-ms** [comment]
> > it seems we left out some providers in common. Is this intentional?
> 
> What are you referring to here?

sorry I remember we talked about how we can leverage common to decouple these providers: 
image
I guess I was mainly referring to localization and accessibility, since these two are nearly identical for both chat and calling

File: AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/StringConstants.swift [hunk 7]
```diff
@@ -5,7 +5,7 @@
...
+enum StringConstants {
     static let defaultEmptyName: String = "Unnamed participant"
     static let localParticipantNamePostfix: String = "(me)"
```
Comment (jimchou-dev) [inline AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/StringConstants.swift:10]: Are these keys being used anywhere maybe we can clean these up?? I think most of the strings should be using LocalizationKey except for the url

**Review by jimchou-dev — DISMISSED** [review]
LGTM

**Comment by emlynmac** [comment]
> 

They need to be refactored; there are tasks for later.

File: AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/StringConstants.swift (hunk 7 above)
Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/StringConstants.swift]: Yeah we can clean those up now.

**Review by jpeng-ms — APPROVED** [review]

**Review by vhuseinova-msft — APPROVED** [review]

**Pull request merged**